runtime/lua/vim/secure.lua

```diff
@@ -169,13 +212,10 @@ function M.trust(opts)
   local trust = read_trust()
 
   if action == 'allow' then
-    local newline = vim.bo[bufnr].fileformat == 'unix' and '\n' or '\r\n'
-    local contents =
-      table.concat(vim.api.nvim_buf_get_lines(bufnr --[[@as integer]], 0, -1, false), newline)
-    if vim.bo[bufnr].endofline then
-      contents = contents .. newline
+    local contents, hash = read_and_hash(fullpath, bufnr)
+    if not contents then
+      return false, string.format("could not read path: %s", fullpath)
```

Comment:
Docstring doesn't mention that this function can return `false` along with an error. It should be updated to include that.